src/test/java/com/juniorstart/juniorstart/service/TechnologyServiceTest.java

```diff
@@ -0,0 +1,165 @@
+        createdAt = new SimpleDateFormat("yyyy-MM-dd").parse("2020-12-31").toInstant();
+        users = new ArrayList<>();
+        users.add(user);
+        technologies.add(technology);
+        page = MockUtil.createMockPage(technologies);
+        userPrincipal = UserPrincipal.create(user);
+        technologyRequest = TechnologyRequest.builder().technologyType(technology.getTechnologyType())
+                .description(technology.getDescription()).title(technology.getTitle()).build();
+        updateTechnologyRequest = UpdateTechnologyRequest.builder().technologyId(technology.getId())
+                .technologyType(technology.getTechnologyType()).description(technology.getDescription())
+                .title(technology.getTitle()).build();
+    }
+
+    @Test
+    public void should_findAll() throws Exception {
+        Mockito.when(technologyRepository.findAll(ArgumentMatchers.isA(Pageable.class))).thenReturn(page);
+        Assert.assertTrue(technologyService.findAll(0, 10).getContent().get(0).getDescription().contains(technology.getDescription()));
+    }
+
+    @Test
+    public void should_addToProject() throws Exception {
+        Mockito.when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        Mockito.when(projectRepository.findById(ArgumentMatchers.anyLong())).thenReturn(Optional.of(project));
+        Mockito.when(technologyRepository.findById(ArgumentMatchers.anyLong())).thenReturn(Optional.of(technology));
+        Assert.assertEquals(technologyService.addToProject(project.getId(), technology.getId(), userPrincipal).getStatusCode(), HttpStatus.OK);
+    }
+
+    @Test
+    public void should_deleteFromProject() throws Exception {
+        Mockito.when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        Mockito.when(technologyRepository.findById(ArgumentMatchers.anyLong())).thenReturn(Optional.of(technology));
+        Mockito.when(projectRepository.findById(ArgumentMatchers.anyLong())).thenReturn(Optional.of(project));
+        Assert.assertEquals(technologyService.deleteFromProject(project.getId(), technology.getId(), userPrincipal).getStatusCode(), HttpStatus.OK);
+    }
+
+    @Test
+    public void should_findById() throws Exception {
+        Mockito.when(technologyRepository.findById(ArgumentMatchers.anyLong())).thenReturn(Optional.of(technology));
+        Assert.assertTrue(technologyService.findById(technology.getId()).getDescription().contains(technology.getDescription()));
+    }
+
+    @Test
+    public void should_createTechnology() throws Exception {
+        Mockito.when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        Mockito.when(technologyRepository.save(ArgumentMatchers.any(Technology.class))).thenReturn(technology);
+        Assert.assertTrue(technologyService.createTechnology(userPrincipal, technologyRequest).getDescription().contains(technology.getDescription()));
+    }
```

Comment:
Po co Mokito. i Assert. jak to są statyczne metody? gdzie verify ?

Reply:
Miałem zamiar dodać po sprawdzeniu.